pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -477,8 +477,12 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings)
 		}
 		if cluster.Http2ProtocolOptions != nil {
 			// This is HTTP/2 in-mesh cluster, advertise it with ALPN.
-			cluster.TlsContext.CommonTlsContext.AlpnProtocols = ALPNInMeshH2
-		} else {
+			if tls.Mode == networking.TLSSettings_ISTIO_MUTUAL {
+				cluster.TlsContext.CommonTlsContext.AlpnProtocols = ALPNInMeshH2
+			} else {
+				cluster.TlsContext.CommonTlsContext.AlpnProtocols = ALPNH2Only
```

Comment:
I don't think we can say ALPNH2only when non istio mutual TLS is used. We need to allow h2 and http/1.1. i.e. the same stuff as the ListenerALPNProtocols

Reply:
Oh ignore.. I see the cluster.Http2ProtocolOptions if check above..